client/src/modules/profiles/component/ProfileForm.tsx

```diff
@@ -0,0 +1,85 @@
+import { Button } from '@chakra-ui/react';
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { Input } from '../../../components/Form/Input';
+import { TextArea } from '../../../components/Form/TextArea';
+import { Form } from '../../../components/Form/Form';
+import { UpdateMeMutationVariables } from '../../../generated/graphql';
+
+interface ProfileFormProps {
+  loading: boolean;
+  onSubmit: (data: UpdateMeMutationVariables) => Promise<void>;
+  data: UpdateMeMutationVariables;
+  submitText: string;
+  loadingText: string;
+}
+
+type Fields = {
+  key: keyof UpdateMeMutationVariables;
+  placeholder: string;
+  label: string;
+  required: boolean;
+  type: string;
+};
+
+const fields: Fields[] = [
+  {
+    key: 'data',
+    label: 'New name',
+    placeholder: 'Please type your new name here',
+    required: true,
+    type: 'text',
+  },
+];
```

Comment:
I couldn't put `key: name` because I was using `key: keyof updateMemutationVariables`. I tried to change it with no luck